From d4fc3cf52c7425dedd7b5daf048aee6bef347bce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pawe=C5=82=20Owoc?= <frut3k7@gmail.com>
Date: Fri, 30 Oct 2026 20:52:54 +0100
Subject: [PATCH] smartdns-ui: fix cross-compilation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove bindgen-cli and rustup checks that fail or attempt to install
tools on the host during make evaluation. Also remove the RUSTFLAGS
override in cargo build so that OpenWrt target flags (including sysroot)
are properly preserved.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
---
 plugin/smartdns-ui/Makefile | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

--- a/plugin/smartdns-ui/Makefile
+++ b/plugin/smartdns-ui/Makefile
@@ -69,24 +69,6 @@ ifneq ($(CARGO_BUILD_TARGET),)
   ARCH_TARGET_PATH=$(CARGO_BUILD_TARGET)/
 endif
 
-# check and install bindgen
-IS_BINDGEN_INSTALL=$(shell which bindgen 2>/dev/null)
-ifeq ($(IS_BINDGEN_INSTALL),)
-  $(shell cargo install --force --locked bindgen-cli)
-endif
-
-ifneq ($(CARGO_BUILD_TARGET),)
-  CARGO_BUILD_TARGET_INSTALL=$(shell rustup target list | grep $(CARGO_BUILD_TARGET) | grep installed)
-  ifeq ($(CARGO_BUILD_TARGET_INSTALL),)
-    # install target
-    $(shell rustup target add $(CARGO_BUILD_TARGET))
-  endif
-endif
-
-IS_MUSL=$(shell $(CC) -v 2>&1 | grep -i musl >/dev/null 2>&1 && echo 1 || echo 0)
-ifeq ($(IS_MUSL), 1)
-override CARGO_RUSTFLAGS +=-C target-feature=-crt-static
-endif
 
 ifdef DEBUG
   CARGO_BUILD_PATH=target/$(ARCH_TARGET_PATH)debug
@@ -111,7 +93,7 @@ test-prepare:
 	$(MAKE) WITH_ZLIB=0 -B -C $(SMARTDNS_SRC_DIR) libsmartdns-test.a
 
 $(BIN):
-	CXXFLAGS= CFLAGS= MAKEFLAGS= RUSTFLAGS="$(CARGO_RUSTFLAGS) $(RUSTFLAGS)" $(CARGO_BUILD_ENV) cargo build $(CARGO_BUILD_ARGS) --features "build-release"
+	CXXFLAGS= CFLAGS= MAKEFLAGS= $(CARGO_BUILD_ENV) cargo build $(CARGO_BUILD_ARGS) --features "build-release"
 	cp $(CARGO_BUILD_PATH)/libsmartdns_ui.so target/smartdns_ui.so
 
 install: $(BIN)
